mne/time_frequency/tests/test_spectrum.py

```diff
@@ -171,6 +171,18 @@ def test_spectrum_reject_by_annot(raw):
     assert spect_no_annot != spect_reject_annot
 
 
+def test_spectrum_reject_by_exclude(raw):
+    """Test bads are not removed unless exclude="bads"."""
+    with raw.info._unlock():
+        raw.pick("mag")
+        if raw.ch_names != []:
+            raw.info["bads"] = raw.ch_names[0:1]
+    spect_no_excld = raw.compute_psd()
+    spect_with_excld = raw.compute_psd(exclude="bads")
+    assert raw.info["bads"] == spect_no_excld.info["bads"]
+    assert raw.info["bads"] != spect_with_excld.info["bads"]
```

Comment:
a better test here would be 

```suggestion
    assert spect_with_excld.info["bads"] == []
```